tracer/src/Datadog.Trace/OTelMetrics/OtlpExporter.cs

```diff
@@ -0,0 +1,460 @@
+        {
+            if (body.Length < 5)
+            {
+                return body;
+            }
+
+            var compressed = body[0] == 1;
+            var len = BinaryPrimitives.ReadUInt32BigEndian(body.Slice(1, 4));
+
+            if (compressed || body.Length < 5 + len)
+            {
+                return body; // We don't handle compressed; return original
+            }
+
+            return body.Slice(5, (int)len);
+        }
+
+        private async Task<bool> SendOtlpRequest(IReadOnlyList<MetricPoint> metrics)
+        {
+            try
+            {
+                // Serialize metrics to protobuf format
+                // Note: JSON serialization is not yet implemented, so we always use protobuf ATM
+                var otlpPayload = _serializer.SerializeMetrics(metrics);
+
+                return _protocol switch
+                {
+                    Configuration.OtlpProtocol.HttpProtobuf => await SendHttpProtobufRequest(otlpPayload).ConfigureAwait(false),
+                    Configuration.OtlpProtocol.HttpJson => await SendHttpJsonRequest(otlpPayload).ConfigureAwait(false),
+                    Configuration.OtlpProtocol.Grpc => await SendGrpcRequest(otlpPayload).ConfigureAwait(false),
+                    _ => await SendHttpProtobufRequest(otlpPayload).ConfigureAwait(false)
+                };
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error sending OTLP request.");
+                return false;
+            }
+        }
+
+        private async Task<bool> SendHttpJsonRequest(byte[] otlpPayload)
+        {
+            Log.Warning("HTTP/JSON protocol is not yet implemented, falling back to HTTP/protobuf");
+            return await SendHttpProtobufRequest(otlpPayload).ConfigureAwait(false);
+        }
+
+        private async Task<bool> SendHttpProtobufRequest(byte[] otlpPayload)
+        {
+            // For UDS, HttpRequestMessage needs http://localhost + path (the socket is chosen by ConnectCallback)
+            // Config already includes /v1/metrics in the endpoint, we just need to rewrite the scheme/host
+            var uri = _endpoint.Scheme == "unix"
+                ? new Uri("http://localhost" + _endpoint.AbsolutePath)
+                : _endpoint;
+
+            using var content = new ByteArrayContent(otlpPayload);
+            content.Headers.ContentType = new MediaTypeHeaderValue("application/x-protobuf");
+
+            using var httpRequest = new HttpRequestMessage(HttpMethod.Post, uri)
+            {
+                Content = content
+            };
+
+            foreach (var header in _headers)
+            {
+                httpRequest.Headers.TryAddWithoutValidation(header.Key, header.Value);
+            }
+
+            return await SendWithRetry(httpRequest).ConfigureAwait(false);
+        }
+
+        private async Task<bool> SendGrpcRequest(byte[] exportRequest)
```

Comment:
The more I think about it, the more I think that we should exclude the gRPC component from this PR. Even though the gRPC protocol is opt-in, I don't think we should be rolling our own gRPC client code as I expect it to be very unreliable. Can we move the gRPC implementation into its own PR so we can merge this PR as soon as we validate the end-to-end `http/protobuf` support?